### ISSUE TITLE
Another Accidental Regression of C* 1.2 Support

### ISSUE THREAD

**Issue by clockfort** [palantir/atlasdb#1883]
Can't use new persistent lock stuff / CAS on old C*, and for whatever reason we're not falling back to the older codepaths where these aren't used.

(I ran into this issue while testing an upgrade for moving a stack from 1.2 to 2.2 C*, though, so there is light at the end of this tunnel, I swear!)

```
Caused by: com.palantir.common.exception.PalantirRuntimeException: Invalid method name: 'cas'
        at com.palantir.common.base.Throwables.createPalantirRuntimeException(Throwables.java:100)
        at com.palantir.common.base.Throwables.createPalantirRuntimeException(Throwables.java:92)
        at com.palantir.common.base.Throwables.throwUncheckedException(Throwables.java:88)
        at com.palantir.atlasdb.keyvalue.cassandra.CassandraTimestampBoundStore.cas(CassandraTimestampBoundStore.java:162)
        at com.palantir.atlasdb.keyvalue.cassandra.CassandraTimestampBoundStore.access$200(CassandraTimestampBoundStore.java:52)
        at com.palantir.atlasdb.keyvalue.cassandra.CassandraTimestampBoundStore$1.apply(CassandraTimestampBoundStore.java:106)
        at com.palantir.atlasdb.keyvalue.cassandra.CassandraTimestampBoundStore$1.apply(CassandraTimestampBoundStore.java:88)
        at com.palantir.atlasdb.keyvalue.cassandra.CassandraClientPoolingContainer.runWithGoodResource(CassandraClientPoolingContainer.java:119)
        at com.palantir.atlasdb.keyvalue.cassandra.CassandraClientPoolingContainer.runWithPooledResource(CassandraClientPoolingContainer.java:96)
        at com.palantir.atlasdb.keyvalue.cassandra.CassandraClientPool.runWithPooledResourceRecordingMetrics(CassandraClientPool.java:633)
        at com.palantir.atlasdb.keyvalue.cassandra.CassandraClientPool.runWithRetryOnHost(CassandraClientPool.java:595)
        at com.palantir.atlasdb.keyvalue.cassandra.CassandraClientPool.runWithRetry(CassandraClientPool.java:572)
        at com.palantir.atlasdb.keyvalue.cassandra.CassandraTimestampBoundStore.getUpperLimit(CassandraTimestampBoundStore.java:88)
        at com.palantir.timestamp.PersistentUpperLimit.<init>(PersistentUpperLimit.java:41)
        at com.palantir.timestamp.PersistentUpperLimit.<init>(PersistentUpperLimit.java:46)
        at com.palantir.timestamp.PersistentTimestampService.create(PersistentTimestampService.java:45)
        at com.palantir.atlasdb.cassandra.CassandraAtlasDbFactory.createTimestampService(CassandraAtlasDbFactory.java:64)
        at com.palantir.atlasdb.factory.ServiceDiscoveringAtlasSupplier.lambda$new$2(ServiceDiscoveringAtlasSupplier.java:68)
        at com.palantir.atlasdb.factory.ServiceDiscoveringAtlasSupplier.getTimestampService(ServiceDiscoveringAtlasSupplier.java:86)
        at com.palantir.atlasdb.services.LockAndTimestampModule.lambda$provideLockAndTimestampServices$1(LockAndTimestampModule.java:38)
        at com.palantir.atlasdb.factory.TransactionManagers.createRawEmbeddedServices(TransactionManagers.java:411)
        at com.palantir.atlasdb.factory.TransactionManagers.createRawServices(TransactionManagers.java:341)
        at com.palantir.atlasdb.factory.TransactionManagers.createLockAndTimestampServices(TransactionManagers.java:312)
        at com.palantir.atlasdb.factory.TransactionManagers.createLockAndTimestampServices(TransactionManagers.java:291)
        at com.palantir.atlasdb.services.LockAndTimestampModule.provideLockAndTimestampServices(LockAndTimestampModule.java:34)
        at com.palantir.atlasdb.services.LockAndTimestampModule_ProvideLockAndTimestampServicesFactory.get(LockAndTimestampModule_ProvideLockAndTimestampServicesFactory.java:2                          2)
        at com.palantir.atlasdb.services.LockAndTimestampModule_ProvideLockAndTimestampServicesFactory.get(LockAndTimestampModule_ProvideLockAndTimestampServicesFactory.java:8                          )
        at com.palantir.atlasdb.shaded.dagger.internal.ScopedProvider.get(ScopedProvider.java:46)
        at com.palantir.atlasdb.services.LockAndTimestampModule_ProvideTimestampServiceFactory.get(LockAndTimestampModule_ProvideTimestampServiceFactory.java:23)
        at com.palantir.atlasdb.services.LockAndTimestampModule_ProvideTimestampServiceFactory.get(LockAndTimestampModule_ProvideTimestampServiceFactory.java:9)
        at com.palantir.atlasdb.shaded.dagger.internal.ScopedProvider.get(ScopedProvider.java:46)
        at com.palantir.atlasdb.services.KeyValueServiceModule_ProvideWrappedKeyValueServiceFactory.get(KeyValueServiceModule_ProvideWrappedKeyValueServiceFactory.java:29)
        at com.palantir.atlasdb.services.KeyValueServiceModule_ProvideWrappedKeyValueServiceFactory.get(KeyValueServiceModule_ProvideWrappedKeyValueServiceFactory.java:9)
        at com.palantir.atlasdb.shaded.dagger.internal.ScopedProvider.get(ScopedProvider.java:46)
        at com.palantir.atlasdb.services.DaggerAtlasDbServices.getKeyValueService(DaggerAtlasDbServices.java:82)
        at com.palantir.atlasdb.performance.benchmarks.table.ConsecutiveNarrowTable.setup(ConsecutiveNarrowTable.java:97)
        at com.palantir.atlasdb.performance.benchmarks.generated.KvsDeleteBenchmarks_allRangeDelete_jmhTest._jmh_tryInit_f_regeneratingcleannarrowtable1_G(KvsDeleteBenchmarks_                          allRangeDelete_jmhTest.java:754)
        at com.palantir.atlasdb.performance.benchmarks.generated.KvsDeleteBenchmarks_allRangeDelete_jmhTest.allRangeDelete_SampleTime(KvsDeleteBenchmarks_allRangeDelete_jmhTes                          t.java:418)
        at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
        at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
        at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
        at java.lang.reflect.Method.invoke(Method.java:498)
        at org.openjdk.jmh.runner.BenchmarkHandler$BenchmarkTask.call(BenchmarkHandler.java:430)
        at org.openjdk.jmh.runner.BenchmarkHandler$BenchmarkTask.call(BenchmarkHandler.java:412)
        at java.util.concurrent.FutureTask.run(FutureTask.java:266)
        at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1142)
        at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:617)
        at java.lang.Thread.run(Thread.java:745)
Caused by: org.apache.thrift.TApplicationException: Invalid method name: 'cas'
        at org.apache.thrift.TApplicationException.read(TApplicationException.java:111)
        at org.apache.thrift.TServiceClient.receiveBase(TServiceClient.java:71)
        at org.apache.cassandra.thrift.Cassandra$Client.recv_cas(Cassandra.java:982)
        at org.apache.cassandra.thrift.Cassandra$Client.cas(Cassandra.java:964)
        at com.palantir.atlasdb.keyvalue.cassandra.CassandraTimestampBoundStore.cas(CassandraTimestampBoundStore.java:152)
        ... 44 more
```

**Comment by gsheasby** [comment]
@clockfort 
Do we need to fix this for the large internal LTS? (0.39.x)?
Now that we're beyond that stage, can we safely drop support for 1.2?

**Comment by clockfort** [comment]
I started an email chain to see whether that is possible. I anticipate some push back, and we might need to keep 1.2 support on atlas develop as well. 
¯\_(ツ)_/¯

**Comment by gsheasby** [comment]
"Resolved" by just removing Cassandra 1.2 support.
